### PR TITLE
Mining fixes.

### LINKED ISSUE #812: Ore processor creates duplicate entries when toggled.
#### Description of issue
https://cdn.discordapp.com/attachments/678827927755816970/746091178386259968/unknown.png
https://cdn.discordapp.com/attachments/678827927755816970/746091188926414970/unknown.png
Images courtesy of Alceris.

#### Difference between expected and actual behavior
Duplicate entries are created, and I think it might not process? I'm not sure.

#### Steps to reproduce
1. Load ore into ore processor.
2. Click to toggle ore.
3. A duplicate entry is created. The original entry is unaffected.

#### Specific information for locating

N/A

#### Length of time in which bug has been known to occur

Unsure, but I have a fix PR up.

#### Client version, Server revision & Game ID

N/A, affects master branch.

#### Issue bingo

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Alternative to #811, fixes #812. Also fixes the stacker, refactors consoles to work nicer, and fixes the tradeship mining area on the map.

### CODE REVIEW

File: code/game/objects/items/weapons/circuitboards/machinery/mining.dm [hunk 1]
```diff
@@ -1,6 +1,11 @@
-/obj/item/stock_parts/circuitboard/mineral_processing
+/obj/item/stock_parts/circuitboard/mineral_processing_console
 	name = T_BOARD("mineral processing console")
-	build_path = /obj/machinery/computer/mining
+	build_path = /obj/machinery/computer/mining/processing
+	origin_tech = "{'programming':2,'engineering':2}"
+
+/obj/item/stock_parts/circuitboard/mineral_stacking_console
+	name = T_BOARD("mineral processing console")
```
Comment (afterthought2) [inline code/game/objects/items/weapons/circuitboards/machinery/mining.dm]: Should change the name if not using buildtype_select.

**Review by afterthought2 — COMMENTED** [review]
While you're at it, I believe there was some change/fix to their density on bay, maybe as they are mapped on walls sometime? Unsure. May be worth checking out.

File: code/game/objects/items/weapons/circuitboards/machinery/mining.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/items/weapons/circuitboards/machinery/mining.dm]: Re: density, they are currently dense and I'm mapping them on a floor, so just going to leave them. When I redo these machines the consoles won't exist anymore.

**Review by afterthought2 — APPROVED** [review]

File: code/modules/mining/machinery/mineral_processor.dm [hunk 2]
```diff
@@ -151,15 +152,17 @@
...
-			if("Nothing")     choice = ORE_DISABLED
-			if("Smelting")    choice = ORE_SMELT
-			if("Compressing") choice = ORE_COMPRESS
-			if("Alloying")    choice = ORE_ALLOY
-		ores_processing[href_list["toggle_smelting"]] = choice
-		. = TRUE
+		var/decl/material/ore_mat = locate(href_list["toggle_smelting"])
+		if(istype(ore_mat) && (ore_mat.type in ores_processing))
+			var/choice = input("What setting do you wish to use for processing [ore_mat.solid_name]?") as null|anything in list("Smelting","Compressing","Alloying","Nothing")
+			if(!choice) return
```
Comment (PsiOmegaDelta) [inline code/modules/mining/machinery/mineral_processor.dm:158]: Don't want to be a stickler when it's mostly rearranged code but as usual with inputs
`/obj/machinery/mineral/processing_unit/Topic(href, href_list, state)`
`if(!choice || !CanInteract(usr, state))`

Comment (MistakeNot4892) [reply code/modules/mining/machinery/mineral_processor.dm:158]: The issue is that this machine may or may not be adjacent to the user as the console is what sends the href.
I'm intending to completely rewrite these machines to remove the console after this goes in so I can sort it out then if you don't mind leaving the crappy code in for awhile.

**Pull request closed without merging**